### PR TITLE
Revisit Release notes for sshlib removal

### LINKED ISSUE #12: Migration from 0.1.45 to 0.2.0 is nightmare
Docs / Release notes wasn't sufficient for 0.2.0 (maybe it would be better to even bump restraint to 1.0.0 due to changes).

Well. We have to learn from mistakes.
So following this has to be addressed.
- Outline fully the scope of the changes for libssh removal in the changelog. We basically destroyed API and backward compatibility. 0.1.45 can't speak with 0.2.0. Port can't be used anymore.
- Revisit all parts of docs and update it accordingly
- Check [BZ](https://bugzilla.redhat.com/show_bug.cgi?id=1821342)

### PR DESCRIPTION
**Pull request by cbouchar**
Providing more details in release notes and documentation to
better advise customer of the impact of this changeset.

Fix #12

### CODE REVIEW

**Comment by renanrodrigo** [comment]
Also, keep the fix line in the commit footer so the issue is closed automatically when the commit is merged

**Comment by cbouchar** [comment]
I had asked Martin about this and there was a disconnect in the conversation.  Do you have any issues with the code itself?

**Comment by StykMartin** [comment]
> Also, keep the fix line in the commit footer so the issue is closed automatically when the commit is merged

You can keep it only in PR.

**Comment by StykMartin** [comment]
To clarify this.
Commits -> Same as before, only difference is that multiple commits will result under one hood. Not like in Gerrit.
During the creation of Pull request (this for) you should reference Issue or BZ. In this case you can edit header of this PR and put ther Fixes: and issue

File: docs/commands.rst [hunk 1]
```diff
@@ -544,6 +544,35 @@ remote test machine.  You can run jobs on the local machine but it is not
...
+.. option:: --job </yourdir/your-job.xml>
+
+   File Location of your job.xml.
+
+.. option:: --restraint-path </dir/restraintd>
+
+   The optional argument ``--restraint-path`` passes the directory from which to run
+   `restraintd` from.  This can be used by developers where the
+   restraint repo is pulled and ``restraintd`` image is built.  By default, the
+   installed image is executed.
```
Comment (danrodrig) [inline docs/commands.rst:566]: This option is for the path of the restraintd binary to run, binary included. It's not only for the directory. Adding to what was written in the option help,

> specify the path to the restraintd binary to run on the remote machine

**Comment by cbouchar** [comment]
I might have lost your comment Daniel on --restraint-path by selecting 'resolve comment' button.  However, I plan to incorporate those changes.

File: docs/commands.rst [hunk 2]
```diff
@@ -544,6 +544,35 @@ remote test machine.  You can run jobs on the local machine but it is not
...
 
+Arguments for the client are as follows::
+
+    restraint --host <1=host> --job <job.xml> [--restraint-path </dir/restraintd>] [-v]
+
+Where:
+
+.. option:: --host <recipe_id>=[<user>@]<host>
+
+   Set host for a recipe with specific id.
```
Comment (StykMartin) [inline docs/commands.rst:555]: Based on this I still don't know why should I specify recipe_id.

File: docs/commands.rst [hunk 3]
```diff
@@ -544,6 +544,35 @@ remote test machine.  You can run jobs on the local machine but it is not
 recommended since some tasks reboot the system. Hosts are tied to recipe IDs
 inside the job XML.
 
+Arguments for the client are as follows::
+
+    restraint --host <1=host> --job <job.xml> [--restraint-path </dir/restraintd>] [-v]
```
Comment (StykMartin) [inline docs/commands.rst:549]: hardcoded 1?

File: docs/commands.rst (hunk 1 above)
Comment (cbouchar) [reply docs/commands.rst]: done

File: docs/commands.rst (hunk 2 above)
Comment (cbouchar) [reply docs/commands.rst]: Sorry for delay.  I wanted to test it out before answering.  Updated code.

Comment (cbouchar) [reply docs/commands.rst]: See updated documentation.

**Pull request merged**